packages/lit-html/src/lit-html.ts

```diff
@@ -851,25 +901,21 @@ export class NodePart {
     return this._startNode.parentNode!;
   }
 
-  _setValue(value: unknown): void {
-    // TODO (justinfagnani): when setting a non-directive over a directive,
-    // we don't yet clear this._directive.
-    // See https://github.com/Polymer/lit-html/issues/1286
+  _setValue(value: unknown, directiveParent: DirectiveParent = this): void {
+    value = resolveDirective(this, value, directiveParent);
```

Comment:
The value before resolving the directive and the value after resolving the directive may be different, so in the worst case. you still need to do the `isPrimitive` check twice. If the pre-resolving `isPrimitive === false`, then it will stay false, but if it's true, it may become false as a result of the directive returning a primitive.

So it would be:
```ts
const valueIsPrimitive = isPrimitive(value);
value = resolveDirective(this, value, directiveParent, valueIsPrimitive);
if (valueIsPrimitive || isPrimitive(value)) {
  // ...
```
and 
```ts
const nextDirectiveConstructor = valueIsPrimitive || isPrimitive(value)
  ? undefined
  : ...
```
Worth it?